18F/dol-whd-14c
# Split apiService into multiple files

# Just an idea

This PR doesn't need to be merged unless @davoodharun thinks it's worthwhile.  It's mostly just me looking at how to improve the CodeClimate score.  Ideally the tests would change as well (and could possibly be simplified), but since from the outside, nothing has changed, the current tests are also okay.

If this is a good direction to go, however, I'll tackle modifying the tests.

---

#### What's this PR do?

To help reduce the low [CodeClimate score on apiService.js](https://codeclimate.com/github/18F/dol-whd-14c/DOL.WHD.Section14c.Web/src/modules/services/apiService.js), this PR splits the apiService into multiple files using factory methods for dependency injection.  All tests still pass.

#### How should this be manually tested?

Pull the code, build, and run.  Make sure API calls from the web client still work.

#### Any background context you want to provide?

There are only two especially bad scores on CodeClimate, and one of them is `apiService.js`.  The main issues are method lines-of-code and cyclomatic complexity.  This PR breaks `apiService` into multiple files to reduce the lines of code of the various files and the `exports` method.
